### PR TITLE
Add multi-ref support to ampliconstats and plot-ampliconstats.

### PR DESCRIPTION
**Pull request by jkbonfield**
The stats output format needs to change for multiple references. I chose to do this always (breaking compatibility with the old
format), but added a -S / --single-ref option to output the old style.  Obviously this is not permitted if the BED file contains more
than one reference.  Plot-ampliconstats automatically detects and works with both output styles.

Amplicon numbers monotonically increase across all references. Processing software will need to check the SS headers to identify how many amplicon numbers are present in each reference.  This helps to keep the old and new output formats similar, with the differences restricted to adding reference names to *SS, AMPLICON, *DP_ALL and *DP_VALID lines.

The output has position based data (*DP_ALL / *DP_VALID) has one line per reference.  Amplicon based stats are either one line per amplicon as before (*TCOORD, AMPLICON, etc) or for the output types that had amplicon numbers as columns there are now more columns covering all amplicons on all references (eg READS, RPERC, DEPTH, etc). Percentages are also computed across all amplicons, not just all amplicons for that reference.

In essence it's pretending to concatenate the references together before generating those reports.  This works well for the plotting for the case where we have many references each with 1 or 2 amplicons (eg genotyping by sequencing).

### CODE REVIEW

**Comment by jkbonfield** [comment]
Bah, I found some bugs immediately after making the PR.  Typical.  I'll let you know when I want it reviewed. :)

**Review by whitwham — COMMENTED** [review]
Looking good.  Just a few very minor points.

File: amplicon_stats.c [hunk 1]
```diff
@@ -75,23 +75,50 @@ typedef struct {
...
+    double  *nrperc, *nrperc2;          // [max_amp]
+    int64_t *nbases, *nbases2;          // [max_amp]
+    int64_t *coverage;                  // [max_amp][max_amp_len]
+    double  (*covered_perc)[MAX_DEPTH]; // [max_amp][MAX_DEPTH]
+    double  (*covered_perc2)[MAX_DEPTH];// [max_amp][MAX_DEPTH];
+    khash_t(tcoord) **tcoord;           // [max_amp+1]
+
+    // 0 is correct pair, 1 is incorrect pair, 2 is unidentified
+    int     (*amp_dist)[3];             // [MAX_AMP][3];
+    //int     amp_pair[MAX_AMP][MAX_AMP]; // dotplot style
```
[the remark this answers is not in the document]
Comment (jkbonfield) [reply amplicon_stats.c:112]: Obviously no.  I meant to tidy such things up.  Good spot and thanks.

**Comment by jkbonfield** [comment]
Trivial edits made and squashed to a single commit.

**Pull request merged**